### PR TITLE
Fixed #25850 -- Made migrations fail loudly on inconsistent history

### PR DESCRIPTION
**Pull request by Uran198**

### CODE REVIEW

File: django/db/migrations/executor.py [hunk 1]
```diff
@@ -59,6 +59,10 @@ def migration_plan(self, targets, clean_start=False):
             else:
                 for migration in self.loader.graph.forwards_plan(target):
                     if migration not in applied:
+                        # Check if migrations were applied before their dependencies
+                        if any([x in applied for x in
+                                self.loader.graph.node_map[migration].children]):
```
Comment (charettes) [inline django/db/migrations/executor.py]: You don't need a list comprehension here.

**Comment by soon** [comment]
Shouldn't `[x for x in self.loader.graph.node_map[migration].children if x in applied]` be added to the `InconsistentMigrationHistory` as a field, so a user could easily track incorrectly applied migrations?

**Comment by Uran198** [comment]
It's strange. How could I break max age expiration?

**Comment by soon** [comment]
Seems like a phantom failure for me. `test_max_age_expiration` depends on time and it may sometimes give false-negative results. You may try to rebuild the PR, though I'm not sure if you allowed to do this. [This doc page](https://code.djangoproject.com/wiki/Jenkins) describes the way of interaction with CI. You may want to squash your two commits into one - this should also trigger the build.

**Comment by Uran198** [comment]
Thanks, it seems it really was a phantom failure.

**Comment by shaib** [comment]
This will stop the executor, alright. Will it also stop the creation of new migrations?

**Comment by Uran198** [comment]
Oh, sorry, I missed that part. When I figure out that part I'll reopen pull request.

**Pull request closed without merging**